### PR TITLE
Addition of a generic Line Chart component using Recharts

### PR DESCRIPTION
**Pull request by WolfyUK**
### Context
[AB#196192](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/196192)

### Change proposed in this pull request
Addition of a generic Line Chart component using Recharts. Addition of `<Stat>` component for highlighting specific key/value pair. Refactor of shared chart props.

### Guidance to review 
Run `front-end-components` to see the new chart in action

### Checklist
- [x] Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)
- [x] Your code builds clean without any errors or warnings
- [x] You have run all unit/integration tests and they pass
- [x] Your branch has been rebased onto main
- [x] You have tested by running locally

### CODE REVIEW

File: front-end-components/src/components/charts/stat/component.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  const entry = useMemo(() => {
+    const entry = Object.entries(data)[displayIndex];
+    if (!entry) {
+      return null;
+    }
+
+    return { label: entry[1][seriesLabelField], value: entry[1][valueField] };
+  }, [data, displayIndex, seriesLabelField, valueField]);
+
+  if (!entry) {
```
Comment (jrabbott) [inline front-end-components/src/components/charts/stat/component.tsx:30]: This throw me for a moment - thinking is was rogue. Would it be better made this as a ternary on the main return?

Comment (WolfyUK) [reply front-end-components/src/components/charts/stat/component.tsx:30]: Could do, but in React I find it much more readable just to bomb out as soon as you can with `null` if there is nothing left to render

Comment (WolfyUK) [reply front-end-components/src/components/charts/stat/component.tsx:30]: Happy to add in a comment instead?

Comment (jrabbott) [reply front-end-components/src/components/charts/stat/component.tsx:30]: Yeah, comment is fine.

**Review by jrabbott — APPROVED** [review]

**Pull request merged**